### ISSUE TITLE
Automatically give contributors feedback for documentation changes

### ISSUE THREAD

**Issue by neumantm** [Exawind/nalu-wind#1120]
I think it would be very helpful to automatically give contributors feedback about documentation changes.

First, it would be nice if the docs of any PR would be build automatically, such that you can see the result.
This would also lower the barrier of entry for new contributors, as they would not need to install all the tools before contributing to the documentation.

Additionally it would be nice to catch any sphinx warnings.
However, this would require all existing sphinx warnings to be fixed first.
(There are currently around 60 warnings.)

Both could be implemented as a GitHub Actions Job.
However, I think it would be more useful to use the service of readthedocs for this.
That ensures the same environment as when the docs are actually build for publishing.

The first step would be to enable readthedocs builds on PRs: https://docs.readthedocs.io/en/stable/pull-requests.html

The second step (after all warnings have been resolved) is to fail on sphinx warnings: https://docs.readthedocs.io/en/stable/tutorial/index.html#making-warnings-more-visible

**Comment by neumantm** [comment]
I've tested both steps using my own rtd project: [running on pr](https://github.com/neumantm/nalu-wind/pull/2) and [fail on warnings](https://github.com/neumantm/nalu-wind/pull/3)

I could also look into fixing those warnings if you want to proceed with this.

**Comment by neumantm** [comment]
Yes of course, no problem.
Ok I'll start working on fixing the existing warning.

However, the first step would be to simply enable the builds on PRs in the readthedocs web UI.
See https://docs.readthedocs.io/en/stable/pull-requests.html
It would be very useful if you (or someone from the team) could enable that.

**Comment by jrood-nrel** [comment]
https://github.com/sandialabs/spack-manager/blob/fa7cd062d6ce770d2f5d92b210654e1007b33bf5/.github/workflows/docs.yml#L34-L40
This causes the PRs to fail if there are warnings in the documentation. We usually use Github pages for deploying documentation, but we haven't moved Nalu-Wind docs here yet.

**Comment by neumantm** [comment]
Yes, that is one possibility. However that can also easily be archived with readthedocs (https://docs.readthedocs.io/en/stable/tutorial/index.html#making-warnings-more-visible).
This could be done very easily and will already help until the migration to github pages will be done.
